go/ocr2/decryptionplugin/decryption.go

```diff
@@ -269,11 +271,20 @@ func (dp *decryptionPlugin) Report(ctx context.Context, ts types.ReportTimestamp
 			continue
 		}
 
-		// OCR2.0 guaranties 2f+1 observations are from distinct oracles
-		// which guaranties f+1 valid observations and, hence, f+1 valid decryption shares.
-		// Therefore, here it is guaranteed that len(decrShares) > f.
+		// This is a sanity check.
+		// OCR2.0 guarantees 2f+1 observations are from distinct oracles.
+		// which guarantees f+1 valid observations and, hence, f+1 valid decryption shares.
+		// Therefore, here it should be guaranteed that len(decrShares) > f.
+		if len(decrShares) < fPlusOne {
+			dp.logger.Error("DecryptionReporting Report: not enough valid decryption shares after processing all observations, skipping aggregation of decryption shares", commontypes.LogFields{
+				"ciphertextID": id,
+			})
+			continue
+		}
+
 		plaintext, err := tdh2easy.Aggregate(ciphertext, decrShares, dp.genericConfig.N)
 		if err != nil {
+			dp.decryptionQueue.SetResult([]byte(id), nil, fmt.Errorf("cannot aggregate decryption shares: %w", err))
```

Comment:
If I understand correctly: if this error occurs, it would not necessarily be the fault of the user, but some bug or issue with the DON such that key shares cannot be aggregated, right?

If the statement above it correct, then we still need to inform the user with some sort of error message which might inform them to re-attempt their request.

Reply:
If it occurs at this place, it is because the the user provided an invalid ciphertext.
The validity of the key shares is guaranteed here:
https://github.com/smartcontractkit/tdh2/blob/4934857d1c4a6820e317d118ba19dd0f3f6eb57f/go/ocr2/decryptionplugin/decryption.go#L329

Reply:
Thanks for clarifying.  In that case we can inform the user they provided invalid ciphertext in the error.